config/webpack.config.dev.js

```diff
@@ -142,18 +149,11 @@ module.exports = {
               name: 'static/media/[name].[hash:8].[ext]',
             },
           },
-          // Process JS with Babel.
+          // Compile .tsx?
           {
-            test: /\.(js|jsx)$/,
+            test: /\.(ts|tsx)$/,
             include: paths.appSrc,
-            loader: require.resolve('babel-loader'),
-            options: {
-              
-              // This is a feature of `babel-loader` for webpack (not Babel itself).
-              // It enables caching results in ./node_modules/.cache/babel-loader/
-              // directory for faster rebuilds.
-              cacheDirectory: true,
-            },
+            loader: require.resolve('ts-loader'),
```

Comment:
I'd suggest using https://github.com/s-panferov/awesome-typescript-loader instead

Reply:
Ok, this is based on https://github.com/Microsoft/TypeScript-React-Starter#typescript-react-starter

Reply:
In my opinion the atl goes better with babel. But, I've not played with it more than one project so the opinion is a _bit_ narrow.

Reply:
Let's keep it like this for now because it does work.

Reply:
I'd add `options: {transpileOnly: true}` so type errors don't get in the way.

Reply:
@AriaMinaei where should I add that? We have two different loaders configured in the webpack for TypeScript.

Reply:
This is my loaders for `/\.tsx?$/` files for development: 
```
         {
            test: /\.(tsx?)$/,
            include: paths.appSrc,
            use: [
              {
                loader: require.resolve('babel-loader'),
                options: {
                  babelrc: true,
                  plugins: ['react-hot-loader/babel'],
                },
              },
              {
                loader: require.resolve('ts-loader'),
                options: {
                  // useTranspileModule: true,
                  transpileOnly: true,
                },
              },
            ],
          },
```